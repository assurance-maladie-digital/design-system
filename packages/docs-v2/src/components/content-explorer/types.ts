export interface TreeviewItem {
    label: string;
    path: string;
    icon?: string;
    children?: TreeviewItem[]
}