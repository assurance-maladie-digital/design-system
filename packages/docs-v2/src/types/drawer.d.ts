interface PageItem {
	title: string;
	icon?: string;
	href?: string;
	to?: string;
}

export interface DrawerItem {
	title: string;
	icon: string;
	items: PageItem[];
}
