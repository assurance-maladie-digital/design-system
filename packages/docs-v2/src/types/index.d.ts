import { IndexedObject } from '@cnamts/vue-dot/src/types';

export interface ApiProp {
	name: string;
	type: string | string[];
	description: string;
	required?: boolean;
	default?: string | number | boolean | null | undefined;
	signature: string;
	value: string;
}

type ItemValue = Pick<ApiProp, 'name' | 'type' | 'default' | 'required' | 'description'>;
type ItemDescription = Pick<ApiProp, 'name' | 'description'>;
type ItemEvent = Pick<ApiProp, 'name' | 'description' | 'value'>;
type ItemFunction = Pick<ApiProp, 'name' | 'signature' | 'description'>;
type ItemModifier = Pick<ApiProp, 'name' | 'type' | 'description'>;
type ItemArgument = Pick<ApiProp, 'type' | 'description'>;

interface ApiHeaderList {
	options: ItemValue[];
	slots: ItemDescription[];
	props: ItemValue[];
	events: ItemEvent[];
	functions: ItemFunction[];
	modifiers: ItemModifier[];
	argument: ItemArgument[];
}

export type HeaderList = keyof ApiHeaderList;
export type ApiHeaders = Partial<ApiHeaderList>;

export type Api = IndexedObject<ApiHeaders>;
