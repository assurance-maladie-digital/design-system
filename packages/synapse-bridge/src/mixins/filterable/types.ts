import type { ChipItem } from '@/elements/ChipList/types';

export interface FilterItem {
  name: string
  value?: unknown
  formatChip?: (value: any) => ChipItem[]
  chipOverflowLimit?: number
  title?: string
  [key: string]: unknown
}

export { type ChipItem };
