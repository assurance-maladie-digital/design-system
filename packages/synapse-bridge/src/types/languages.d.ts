declare module 'languages' {
	import type { Language } from './types.d.ts';

	export function getLanguageInfo(langCode: string): Language;
	export function getAllLanguageCode(): string[];
};
