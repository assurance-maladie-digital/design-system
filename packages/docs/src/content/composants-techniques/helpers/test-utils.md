---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
