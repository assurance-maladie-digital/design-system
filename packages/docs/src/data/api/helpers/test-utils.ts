import { Api } from '~/types';

export const api: Api = {
	applyGlobalConfig: {
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installGlobalPlugins: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	installRouter: {
		arguments: [
			{
				name: 'localVue',
				description: 'Local Vue instance',
				type: 'Vue',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	createRouter: {
		arguments: [
			{
				name: 'options',
				description: 'Router options',
				type: 'RouterOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'VueRouter',
			},
		],
	},
	createStore: {
		arguments: [
			{
				name: 'options',
				description: 'Store options',
				type: 'StoreOptions',
				required: false,
			},
		],
		returnValue: [
			{
				type: 'Store',
			},
		],
	},
	createVuetifyInstance: {
		arguments: [
			{
				name: 'preset',
				description: 'Vuetify preset',
				type: 'Partial<UserVuetifyPreset>',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'Vuetify',
			},
		],
	},
	mockVFormRef: {
		arguments: [
			{
				name: 'isValid',
				description: 'Is valid',
				type: 'boolean',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
	mockTranslations: {
		arguments: [
			{
				name: 'translations',
				description: 'Translations',
				type: 'IndexedObject',
				required: true,
			},
		],
		returnValue: [
			{
				type: 'void',
			},
		],
	},
};
