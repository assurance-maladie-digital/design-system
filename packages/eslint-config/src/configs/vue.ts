import { type Parser, type FlatESLintConfigItem } from 'eslint-define-config'
import { parserTs, parserVue, pluginVue } from '../plugins'
import { GLOB_VUE } from '../globs'
import { hasTypeScript } from '../env'

export const vue: FlatESLintConfigItem[] = [
	{
		plugins: {
			vue: pluginVue,
		},
	},
	{
		files: [GLOB_VUE],
		languageOptions: {
			parser: parserVue,
			parserOptions: {
				extraFileExtensions: ['.vue'],
				parser: hasTypeScript ? parserTs as unknown as Parser : undefined,
				sourceType: 'module',
			},
		},
		processor: pluginVue.processors['.vue'],
		rules: {
			// TODO: configure
		},
	},
]
