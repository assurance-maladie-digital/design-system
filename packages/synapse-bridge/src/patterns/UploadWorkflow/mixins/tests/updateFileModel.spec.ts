import { defineComponent } from "vue";
import { describe, it, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'

import { FileListItem } from "../../types";

import UpdateFileModel from "../updateFileModel";

const testComponent = defineComponent( {
	name: "TestComponent",
	mixins: [UpdateFileModel],
	template: "<div/>",
});

/** Get list of files */
function getFileList(): FileListItem[] {
	return [
		{
			id: 'file1',
			title: 'UploadWorkflow',
			name: 'avatar.png',
			file: {} as File
		}
	];
}

describe("updateFileModel", () => {
	it("inits the file list", () => {
		const wrapper = shallowMount(testComponent);

		wrapper.vm.initFileList(getFileList());

		expect(wrapper.vm.fileList).toMatchSnapshot();
	});

	it("does not overwrite existing state", () => {
		const wrapper = shallowMount(testComponent);

		const fileListState = getFileList();
		fileListState[0].state = "success";

		wrapper.vm.initFileList(fileListState);

		expect(wrapper.vm.fileList).toMatchSnapshot();
	});

	it("returns undefined if the file does not exists", () => {
		const wrapper = shallowMount(testComponent);

		wrapper.vm.fileList = getFileList();

		expect(
			wrapper.vm.updateFileModel(1, "state", "initial")
		).toBeUndefined();
	});

	it("deletes the key if the value is undefined", () => {
		const wrapper = shallowMount(testComponent);

		wrapper.vm.fileList = getFileList();

		wrapper.vm.updateFileModel(0, "name", undefined);

		expect(wrapper.vm.fileList[0]).toMatchSnapshot();
	});

	it("updates the value of the corresponding key", () => {
		const wrapper = shallowMount(testComponent);

		wrapper.vm.fileList = getFileList();

		wrapper.vm.updateFileModel(0, "state", "updated");

		expect(wrapper.vm.fileList[0].state).toBe("updated");
	});
});
