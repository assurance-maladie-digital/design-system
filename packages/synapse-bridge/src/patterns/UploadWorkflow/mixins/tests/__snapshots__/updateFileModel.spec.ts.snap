// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`updateFileModel > deletes the key if the value is undefined 1`] = `
{
  "file": {},
  "id": "file1",
  "title": "UploadWorkflow",
}
`;

exports[`updateFileModel > does not overwrite existing state 1`] = `
[
  {
    "file": {},
    "id": "file1",
    "name": "avatar.png",
    "state": "success",
    "title": "UploadWorkflow",
  },
]
`;

exports[`updateFileModel > inits the file list 1`] = `
[
  {
    "file": {},
    "id": "file1",
    "name": "avatar.png",
    "state": "initial",
    "title": "UploadWorkflow",
  },
]
`;
